MPT Bug with MAPL 2.0.6 and MAPL 2.1.x

PR #312 triggered a bug in MPT 2.17:
```
/nobackup/mathomp4/Models/GEOSgcm-stock/GEOSgcm/src/Shared/@MAPL/MAPL_Base/MAPL_Cap.F90(589): error #6780: A dummy argument with the INTENT(IN) attribute shall not be defined nor become undefined.   [THIS]
         call MPI_Comm_set_errhandler(this%comm_world,MPI_ERRORS_RETURN,ierror)
--------------------------------------^
compilation aborted for /nobackup/mathomp4/Models/GEOSgcm-stock/GEOSgcm/src/Shared/@MAPL/MAPL_Base/MAPL_Cap.F90 (code 1)
src/Shared/@MAPL/MAPL_Base/CMakeFiles/MAPL_Base.dir/build.make:446: recipe for target 'src/Shared/@MAPL/MAPL_Base/CMakeFiles/MAPL_Base.dir/MAPL_Cap.F90.o' failed
```
This seems due to having an incorrect interface for the `MPI_Comm_set_errhandler` function:
```fortran
    SUBROUTINE MPI_COMM_SET_ERRHANDLER(COMM, ERRHANDLER, IERROR)
    USE MPI_DEFAULT_KIND_M
    IMPLICIT NONE
    INTEGER(KIND=DEFKIND), INTENT(INOUT)    :: COMM
    INTEGER(KIND=DEFKIND), INTENT(IN)    :: ERRHANDLER
    INTEGER(KIND=DEFKIND), INTENT(OUT)   :: IERROR
    END SUBROUTINE
```
compare to the standard for mpi_f08:
```fortran
USE mpi_f08
MPI_Comm_set_errhandler(comm, errhandler, ierror)
    TYPE(MPI_Comm), INTENT(IN) :: comm
    TYPE(MPI_Errhandler), INTENT(IN) :: errhandler
    INTEGER, OPTIONAL, INTENT(OUT) :: ierror
```

This seems fixed in MPT 2.21 at NAS as it seems to use:
```fortran
    SUBROUTINE MPI_COMM_SET_ERRHANDLER(COMM, ERRHANDLER, IERROR)
    USE MPI_DEFAULT_KIND_M
    IMPLICIT NONE
    INTEGER(KIND=DEFKIND), INTENT(IN)    :: COMM
    INTEGER(KIND=DEFKIND), INTENT(IN)    :: ERRHANDLER
    INTEGER(KIND=DEFKIND), INTENT(OUT)   :: IERROR
    END SUBROUTINE
```
but that is not the "recommended" MPT as yet. And more testing would be needed (including new Baselibs).

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,6 +9,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 ### Fixed
+
+- Workaround for MPT 2.17 build bug with `MPI_Comm_set_errhandler`
+
 ### Removed
 ### Added
 

--- a/MAPL_Base/MAPL_Cap.F90
+++ b/MAPL_Base/MAPL_Cap.F90
@@ -653,15 +653,18 @@ contains
       class (KeywordEnforcer), optional, intent(in) :: unusable
       integer, optional, intent(out) :: rc
 
-      integer :: ierror
+      integer :: ierror, local_comm_world
       _UNUSED_DUMMY(unusable)
 
       if (.not. this%mpi_already_initialized) then
 #ifdef BUILD_TYPE_IS_NOT_DEBUG
+         ! MPT 2.17 has a bug with its interface to MPI_Comm_set_errhandler
+         ! defining comm as inout instead of in.
+         local_comm_world = this%comm_world
          ! Intel MPI at NCCS seems to have spurious MPI_Finalize errors that do
          ! not affect the answer or even the finalize step. This call suppresses
          ! the errors.
-         call MPI_Comm_set_errhandler(this%comm_world,MPI_ERRORS_RETURN,ierror)
+         call MPI_Comm_set_errhandler(local_comm_world,MPI_ERRORS_RETURN,ierror)
          _VERIFY(ierror)
 #endif
          call MPI_Finalize(ierror)

